components/clarinet-sdk-wasm/src/core.rs

```diff
@@ -338,33 +348,43 @@ impl SDK {
 
                     let contracts_sources = self.file_accessor.read_files(contracts_paths).await?;
 
-                    let deployment = DeploymentSpecification::from_specifications(
+                    let existing_deployment = DeploymentSpecification::from_specifications(
                         &spec_file,
                         &StacksNetwork::Simnet,
                         &project_root,
                         Some(&contracts_sources),
                     )
                     .map_err(|e| e.to_string())?;
 
-                    let artifacts = setup_session_with_deployment(&manifest, &deployment, None);
+                    let (deployment_with_only_contract_publish_txs, custom_batches) =
+                        existing_deployment.extract_no_contract_publish_txs();
+
+                    let (deployment, artifacts) = if deployment_with_only_contract_publish_txs
+                        == default_deployment
+                    {
+                        log!("{}", "using existing deployment plan".yellow().bold());
+                        let artifacts =
+                            setup_session_with_deployment(&manifest, &existing_deployment, None);
+                        (existing_deployment, artifacts)
+                    } else {
+                        log!("{}", "using updated deployment plan".yellow().bold());
+                        default_deployment.merge_batches(custom_batches);
```

Comment:
> So in both cases, we're deploying the same thing?

Right

> Does this mean that a user is not able to change the contract publish txs and have those be deployed?

Yes, we consider the Clarinet.toml to be se source of truth regarding contract deployments, and the deployment plan follows that.

> a flag to allow a user to deploy `deployment_with_only_contract_publish_txs.merge(custom_batches)`?

So `deployment_with_only_contract_publish_txs.merge(custom_batches)` is simply `existing-deployment`
We could totally consider that 👍 
What this current implementation would be to add a contract-call in between two contract-publish

```
-id: 0
  contract-publish
- id: 1
  contract-call
- id: 2
  - contract-publish
```

It will be rewritten to
```
- id: 0
  contract-publish
  contract-publish
- id: 1


👉 I opened an issue to follow up on that
  contract-call
```